cg/utils/date.py

```diff
@@ -16,20 +16,10 @@
 LOG = logging.getLogger(__name__)
 
 
-def match_date(date):
-    """Check if a string is a valid date
-
-    Args:
-        date(str)
-
-    Returns:
-        bool
-    """
+def match_date(date: str) -> bool:
+    """Check if a string is a valid date"""
```

Comment:
This can do the job and parse most date string formats!
https://dateutil.readthedocs.io/en/stable/parser.html

Reply:
We had this as dependency before and I wrote the module to get rid of that dependency:) It is not well maintained and fragile to have in the dependency tree